src/Packetery/Module/Uninstaller.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Packetery\Module;
+
+use Packetery\Module\Carrier\CountryListingPage;
+use Packetery\Module\Carrier\OptionPrefixer;
+use Packetery\Module\Options\Exporter;
+use Packetery\Module\Options\OptionsProvider;
+use Packetery\Module\Order\CollectionPrint;
+use Packetery\Module\Order\LabelPrint;
+use Packetery\Module\Order\Metabox;
+
+class Uninstaller {
+
+	/**
+	 * @var Options\Repository
+	 */
+	private $optionsRepository;
+
+	/**
+	 * @var WpdbAdapter
+	 */
+	private $wpdbAdapter;
+
+	public function __construct(
+		Options\Repository $optionsRepository,
+		WpdbAdapter $wpdbAdapter
+	) {
+		$this->optionsRepository = $optionsRepository;
+		$this->wpdbAdapter       = $wpdbAdapter;
+	}
+
+	/**
+	 * Uninstalls plugin and drops custom database table.
+	 */
+	public function uninstall(): void {
+		if ( defined( 'PACKETERY_DEBUG' ) && constant( 'PACKETERY_DEBUG' ) === true ) {
+			return;
+		}
+
+		if ( is_multisite() ) {
+			$this->cleanUpForMultisite();
+		} else {
+			$this->cleanUp();
+		}
+	}
+
+	/**
+	 * Drops all plugin tables and options when Multisite is enabled.
+	 */
+	private function cleanUpForMultisite(): void {
+		$sites = Plugin::getSites();
+
+		foreach ( $sites as $site ) {
+			switch_to_blog( $site );
+			$this->cleanUp();
+			restore_current_blog();
+		}
+	}
+
+	/**
+	 * Drops all plugin tables and options for a single site.
+	 */
+	private function cleanUp(): void {
+		$this->wpdbAdapter->query( 'DROP TABLE IF EXISTS `' . $this->wpdbAdapter->packeteryLog . '`' );
+		$this->wpdbAdapter->query( 'DROP TABLE IF EXISTS `' . $this->wpdbAdapter->packeteryCarrier . '`' );
+		$this->wpdbAdapter->query( 'DROP TABLE IF EXISTS `' . $this->wpdbAdapter->packeteryOrder . '`' );
+		$this->wpdbAdapter->query( 'DROP TABLE IF EXISTS `' . $this->wpdbAdapter->packeteryCustomsDeclarationItem . '`' );
+		$this->wpdbAdapter->query( 'DROP TABLE IF EXISTS `' . $this->wpdbAdapter->packeteryCustomsDeclaration . '`' );
+
+		$transientsToDelete = $this->optionsRepository->getAllTransientsByPrefixes(
+			[
+				Checkout::TRANSIENT_CHECKOUT_DATA_PREFIX,
```

Comment:
Co uložit názvy a prefixy transientů do dedikované třídy a tím zajistit, že bude řešení vyhovovat poslední větě zadání?

Reply:
Mnou navrhovaná tříba by významně snížila riziko, že někdo zapomene svůj nový transient přidat do Uninstalleru. Kód uninstalleru by byl hezčí.